packages/react-data-grid/src/Header.tsx

```diff
@@ -15,12 +15,12 @@ interface HeaderRowData extends Pick<HeaderRowProps, 'filterable' | 'onFilterCha
 }
 
 type SharedHeaderRowProps = Pick<HeaderRowProps,
-  'draggableHeaderCell'
-  | 'getValidFilterValues'
-  | 'sortDirection'
-  | 'sortColumn'
-  | 'onHeaderDrop'
-  | 'onSort'
+'draggableHeaderCell'
+| 'getValidFilterValues'
+| 'sortDirection'
+| 'sortColumn'
+| 'onHeaderDrop'
+| 'onSort'
```

Comment:
It'll be fixed eventually, gotta give it time.
https://github.com/typescript-eslint/typescript-eslint/issues/455